### DISCUSSION TITLE
5/15(월) 비동기/네트워크 스터디

### DISCUSSION THREAD

**Discussion by jjunhaa0211** [Info-iOS/iOStudy#139] [Q&A] [answered]
## 시작기간 = 5/16 ~ 5/29

- 1일차
   - 동기/비동기/동시/직렬
- 2일차
  - async await
- 3일차
  - GCD
- 4일차
  - DispatchWorkItem
  - DispatchGroup
- 5일차
  - Semaphore
- 6일차
  - URLSession을 이용한 기본적인 네트워크 학습
  - URLSessionTask, URLSessionDataTask, URLSessionUploadTask 차이점
  - URLSessionConfiguration과 URLSessionDelegate 학습
- 7일차
  - URLRequest를 이용한 HTTP 요청 학습
  - HTTP Method(GET, POST, PUT, DELETE) 학습
  - URLComponents를 이용한 URL 생성과 파라미터 전달 학습
- 8일차
  -  URLSession을 이용한 데이터 다운로드 및 업로드 방법 학습
  - 다운로드한 데이터를 이용하여 이미지를 표시하는 방법 학습
  - URLRequest와 함께 업로드할 데이터를 생성하는 방법 학습
- 9일차
  - URLSession을 이용한 파일 다운로드 및 업로드 방법 학습
  - 파일 다운로드 시 프로그레스 바를 이용한 다운로드 진행 상황을 표시하는 방법 학습
  - 파일 업로드 시 HTTP Body와 함께 전송할 데이터를 생성하는 방법 학습
- 10일차
  - URLSession을 이용한 응답 처리 방법 학습
  - HTTP Status Code에 대한 이해와 처리 방법 학습
  - URLSessionTaskDelegate를 이용한 응답 처리 학습
- 11일차
  - URLSession을 이용한 에러 처리 방법 학습
  - 네트워크 에러 및 HTTP Status Code에 따른 에러 처리 방법 학습
  - URLSessionTaskDelegate와 URLSessionDelegate를 이용한 에러 처리 학습
- 12일차
  - URLSession을 이용한 캐시 처리 방법 학습
  - 캐시를 이용하여 반복적인 네트워크 통신을 줄이는 방법 학습
  - 캐시 정책에 따른 캐시 처리 방법 학습
- 13일차
  - URLSession을 이용한 백그라운드 테스크 처리 방법 학습
  - 백그라운드에서 네트워크 통신을 처리하는 방법 학습
  - URLSessionConfiguration 및 URLSessionDelegate를 이용한 백그라운드 테스크 처리 학습
- 14일차
  - URLSession을 이용한 인증 처리 방법 학습
  - 네트워크 통신 시 HTTP 인증 처리 방법 학습
  - URLSessionDelegate를 이용한 인증 처리 학습
- 15일차
  - 테스트 및 프로젝트 적용 
  
 
## 스터디를 하면서 조건이 몇가지 있습니다.

1. 누가 봐도 이해되게 최대한 쉽게 풀어써주세요
2. 줄은 적어도 20줄은 넘겨주세요
3. 예제 코드를 최대한 쉽게 적어주세요
4. 오늘 해야할 숙제는 무조건 오늘 끝내주세요
5. 커밋하는 방법
  1. code -> https://github.com/Info-iOS/iOStudy.git를 깃크라켄에 클론
  2. 깃허브 레포 권한 요청(제가 권한을 줌)
  3. 브랜치를 판다
    - 이름 규칙
      - EX) URLSession-junha
  4. 파일을 NetworkStudy 에 파일 추가
  5. 커밋 -> 푸시 -> pr 요청
  6. pr 비속어만 안들어가면 ㅇㅋ
  7. 가능하면 다른 사람의 스터디 내용도 한번씩 읽어보면 성장에 도움이 될 것 입니다!! 

## 소통

소통은 기간이 길지 않기 떄문에 왠만하면 깃허브 Discussions에서 그날 올라간 { 0/00(요일) x일차 } 여기서 할 생각입니다.

그리고 하고 싶으신 분 있으면 언제나 하시면 됩니다!!

# 🫠

**Comment by jjunhaa0211** [comment, marked as answer]
하고 싶은 사람은 밑에 아무 이미지를 넣어주세요 

ex) 😏

**Reply by Eunho0922** [reply]
😵‍💫

**Reply by gurdl0525** [reply]
🙃

**Reply by BJCHO0501** [reply]
😁
